### PR TITLE
Add support for M3 motion

### LINKED ISSUE #116525: Support Material 3 motion - Easing and duration
https://m3.material.io/styles/motion/easing-and-duration/applying-easing-and-duration

- [ ] Add support for all curves
- [ ] Add support for all durations

Existing material curves are at https://github.com/flutter/flutter/blob/main/packages/flutter/lib/src/material/curves.dart#L26. We should probably create `motion.dart` and put everything there. Some thought needs to be put into the naming for legacy curves.

### PR DESCRIPTION
**Pull request by guidezpl**
## Description

This adds support for M3 easing and duration tokens.

This PR includes these changes:
* Generation of duration and easing constants, in `Durations` and `Easing`, respectively (`Curves` is already taken in the `animation` library)
* Add 3 Dart fixes

Once this is merged, I'll migrate packages/plugins/customers and then uncomment the deprecation notices for the 3 M2 curves, all of which have 1:1 replacements.

## Related Issues
 - Fixes https://github.com/flutter/flutter/issues/116525

## Tests
 - Added Dart fix tests

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test_fixes/material/material.dart.expect
Comment (guidezpl) [inline packages/flutter/test_fixes/material/material.dart.expect:1]: It seems this may not be possible via Dart fix?

Comment (guidezpl) [reply packages/flutter/test_fixes/material/material.dart.expect:1]: @Piinks this isn't currently supported, should I include the dart fix, but commented out? Or remove it altogether?

Comment (Piinks) [reply packages/flutter/test_fixes/material/material.dart.expect:1]: Is there a dart issue we can link to? If so, I'd go with commenting it out and referencing the issue. Else I would just omit the fix.

Comment (guidezpl) [reply packages/flutter/test_fixes/material/material.dart.expect:1]: Sounds good, commented out and added a TODO

**Comment by guidezpl** [comment]
friendly bump for review :)

**Review by QuncCccccc — APPROVED** [review]
LGTM:)

File: dev/tools/gen_defaults/generated/used_tokens.csv [hunk 1]
```diff
@@ -856,6 +856,31 @@ md.sys.elevation.level2,
...
+md.sys.motion.duration.medium4Ms,
+md.sys.motion.duration.short1Ms,
+md.sys.motion.duration.short2Ms,
+md.sys.motion.duration.short3Ms,
+md.sys.motion.duration.short4Ms,
+md.sys.motion.easing.emphasized.accelerate,
+md.sys.motion.easing.emphasized.decelerate,
+md.sys.motion.easing.legacy,
+md.sys.motion.easing.legacy.accelerate,
+md.sys.motion.easing.legacy.decelerate,
```
Comment (gnprice) [inline dev/tools/gen_defaults/generated/used_tokens.csv:879]: It looks like this doesn't include the token `md.sys.motion.easing.emphasized`, corresponding to what the spec text calls "Emphasized easing". Is that intended?

Comment (guidezpl) [reply dev/tools/gen_defaults/generated/used_tokens.csv:879]: I believe it was added in a later tokens version (we're on 0.162)

**Pull request merged**